repo/ecc/ecc_utils.go

```diff
@@ -0,0 +1,69 @@
+package ecc
+
+import "math"
+
+func ComputeShards(spaceOverhead float32) (required, redundant int) {
+	required = 128
+	redundant = between(applyPercent(required, spaceOverhead/100), 1, 128)
+
+	if redundant == 1 {
+		redundant = 2
+		required = between(applyPercent(redundant, 100/spaceOverhead), 128, 254)
+	}
+
+	// Berlekamp-Welch error correction works better with an even number
+	if redundant%2 == 1 {
+		redundant++
+	}
+
+	return
+}
+
+func between(val int, min int, max int) int {
+	switch {
+	case val < min:
+		return min
+	case val > max:
+		return max
+	default:
+		return val
+	}
+}
+
+func applyPercent(val int, percent float32) int {
+	return int(math.Floor(float64(val) * float64(percent)))
+}
+
+func clear(bytes []byte) {
+	for i := range bytes {
+		bytes[i] = 0
+	}
+}
+
+func minInt(a int, b int) int {
```

Comment:
this seems to be used once or twice, sometimes if statement is more readable

Reply:
I can change it if you prefer but I really think a function called `min` is easier to understand. The `Int` I did not like, but I could not make a generic version that works for all types of numbers.